cmd/provenanced/cmd/root.go

```diff
@@ -64,7 +66,7 @@
 	sdk.SetAddrCacheEnabled(false)
 	defer sdk.SetAddrCacheEnabled(true)
 
-	app.SetConfig(isTestnetFlagSet(), false)
+	app.SetConfig(isTestnetFlagSet(os.Args[1:]), false)
```

Comment:
All the flag/arg/subcommand stuff is defined and handled by `cobra` (which uses, `pflags` for some stuff). As the comment above `isTestnetFlagSet` states, this is a solution to a bootstrap problem. The SDK has made it so that we need to know the value of the `--testnet` boolean flag before we can finish setting up all the commands (and all the flag stuff). So, in this `func init()`, there's no flag definitions yet, which would be needed in order to use something like `pflags`. And I don't want to define the `--testnet` flag here because it needs to be a persistent flag on all the commands; I don't want possibly conflicting definitions, and I can't say for certain that doing so wouldn't interfere with all of the arg processing that `cobra` does.

The `isTestnetFlagSet` manually parses for one single boolean flag that's allowed to be anywhere in the args list regardless of subcommands and other flags and args. There are no complex scenarios or flag combinations that it needs consider.